schemas/manifest.schema.json

```
@@ -0,0 +1,455 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://github.com/YunoHost/apps/blob/master/schemas/manifest.schema.json",
+    "title": "Yunohost app manifest.toml schema",
+    "version": "0",
+
+    "type": "object",
+    "$defs": {
+        "translated_string": {
+            "type": "object",
+            "required": ["en"],
+            "additionalProperties": false,
+            "patternProperties": {
+                "^[a-z]{2}$": {
+                    "type": "string"
+                }
+            }
+        },
+        "byte_size": {
+            "type": "string",
+            "pattern": "^[0-9]*(\\.[0-9]*)?[kKmMgG]$"
+        },
+        "boolean_or_not_relevant": {
+            "anyOf": [
+                {
+                    "type": "boolean"
+                },
+                {
+                    "type": "string",
+                    "const": "not_relevant"
+                }
+            ]
+        },
+        "string_or_array_of": {
+            "anyOf": [
+                {
+                    "type": "string"
+                },
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            ]
+        }
+    },
+
+    "properties": {
+        "packaging_format": {
+            "description": "Packaging format / version",
+            "type": "integer"
+        },
+        "id": {
+            "description": "App ID",
+            "type": "string"
+        },
+        "name": {
+            "description": "App Name",
+            "type": "string"
+        },
+        "version": {
+            "description": "App and package version",
+            "type": "string",
+            "pattern": "^[0-9\\.]*~ynh[0-9]*"
+        },
+        "description": {"$ref": "#/$defs/translated_string"},
+        "maintainers": {
+            "description": "List of maintainers",
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+        },
+        "upstream": {
+            "type": "object",
+            "required": ["license", "website", "code"],
```

```suggestion
            "required": ["license"],
```

Unfortunately we can't make those mandatory because some small projects often have no actual website, and (more rarely), we have apps which have no actual code upstream, such as my_webapp, vpnclient, etc.

---

Alright :)